### PR TITLE
Handle constant B-spline mutual information inputs

### LINKED ISSUE #9018: GlobalMutualInformationLoss returns NaN for constant B-spline inputs
**Describe the bug**

`GlobalMutualInformationLoss(kernel_type="b-spline")` returns a non-finite loss for valid constant prediction or target tensors. Constant tensors have zero intensity range, so the B-spline bin width becomes zero and normalization produces NaNs.

Two related boundary failures exist in the same path:

- `num_bins <= 4` cannot accommodate the implementation's two-bin padding on each side, leaving a zero or negative interior denominator.
- In float16, the cubic B-spline distance polynomial can overflow before masking.

**To Reproduce**

```python
import torch
from monai.losses import GlobalMutualInformationLoss

pred = torch.zeros((1, 1, 8, 8), requires_grad=True)
target = torch.ones_like(pred)
loss = GlobalMutualInformationLoss(kernel_type="b-spline")
result = loss(pred, target)
print(result, torch.isfinite(result))
```

On the current `dev` revision, `result` is non-finite.

**Expected behavior**

Valid constant tensors should produce a finite loss and finite gradients. B-spline configurations with too few bins should fail at construction with a clear `ValueError`.

**Screenshots**

Not applicable.

**Environment**

```text
MONAI version: 1.6.0rc1+46.g3ee058bd
MONAI revision: 3ee058bdd16dd4a566d23d3f84687c3c35268a36
Python version: 3.13.13
PyTorch version: 2.12.0
NumPy version: 2.4.6
System: macOS 27.0 arm64
CUDA available: False
```

**Additional context**

A focused fix and regression coverage are proposed in #9022.

### PR DESCRIPTION
**Pull request by yinkev**
Closes #9018.

### Description

B-spline global mutual information now remains finite for constant prediction or target tensors, rejects bin counts that cannot accommodate the required spline padding, and promotes half-precision spline distances before the cubic basis calculation.

Constant tensors have zero intensity range, so the previous bin width was zero and normalization produced NaNs. Separately, the cubic polynomial could overflow for float16 distances, while `num_bins <= 4` made the padded interior denominator zero or negative.

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### Validation

- float32 and float16 constant-input regressions fail on `upstream/dev` and pass with the fix;
- `QUICKTEST=true python -m pytest tests/losses/image_dissimilarity/test_global_mutual_information_loss.py -q`: 15 passed, 2 skipped;
- `ruff check` on both touched files;
- `git diff --check`.

### CODE REVIEW

**Comment by yinkev** [comment]
@coderabbitai review

**Comment by yinkev** [comment]
Replaced by #9022 to conform to this repository's documented contributor workflow and branch-naming convention. The replacement uses the same signed commit and identical code diff: https://github.com/Project-MONAI/MONAI/pull/9022
